Tests/ValidationTests.swift

```diff
@@ -776,19 +777,19 @@ extension DownloadRequest {
         return validate { (request, response, _) in
             let fileURL = self.fileURL
 
-            guard let validFileURL = fileURL else { return .failure(ValidationError.missingFile) }
+            guard let validFileURL = fileURL else { return .failure(ValidationError.missingFile.error) }
 
             do {
                 let _ = try Data(contentsOf: validFileURL)
                 return .success(Void())
             } catch {
-                return .failure(ValidationError.fileReadFailed)
+                return .failure(ValidationError.fileReadFailed.error)
             }
         }
     }
 
     func validate(with error: Error) -> Self {
-        return validate { (_, _, _) in .failure(error) }
+        return validate { _, _, _ in .failure(error as? AFError ?? .responseValidationFailed(reason: .customValidationFailed(error: error))) }
```

Comment:
Again, better not to have to do this for custom validations.

Reply:
Reverted.